### ISSUE TITLE
storyplanner png "malformed frame name" error

### ISSUE THREAD

**Issue by matitanimata** [opentoonz/opentoonz#2094]
### Issue Summary

unable to load png files exported from storyplanner because of a  "malformed frame name" error

### Steps to Reproduce
create a storyboard in storyplanner 
export it as a toonz project
rename the prj6.xml file as otprj.xml to make it readable in OT
load a scene from the project: it won't load the storyboard images.
### Expected Results
storyplanner is an helpful  tool in a toonz or OT pipeline: with it you can create a storyboard, an animatic and export all as a toonz project. In this way you'll automatically have a sort of layout of each scene with story sketches, soundtrack and camera information.

### Actual Results
in the last versions of OT when I try to load a scene exported from storyplanner I get a  "malformed frame name: skipping frame" message and OT is unable to load the images.

### System Information
OT 1.2.1
win 10
i7
8gb ram
wacom companion 2
![2018-06-21](https://user-images.githubusercontent.com/20012515/41700731-cec08b0c-7529-11e8-88dc-d22c5c4f0d77.png)

**Comment by shun-iwasawa** [comment]
After #1833 was merged on April 23, the restriction on filename for sequential images was relaxed.
Now the file name should not have 4-digit frame numbers (like `A.0001.png` ) and the file `scene_024_001_0.png` is treated as a part of sequential files.
(Any image files with name in a format like `[levelname]["_"or"."][any digits][alphabet(optional)].[extension]` will be treated as sequential.)
And since OT does not support frame number starting from 0 for now, it skip loading the file.

I'm not familiar with Storyplanner, but can you specify the exported file name not to be treated as sequential by OT?

**Comment by matitanimata** [comment]
it names files in this way: Scene_022_001_0.png    022=scene number, 001=panel nu
![2018-06-21 1](https://user-images.githubusercontent.com/20012515/41703538-f31302be-7533-11e8-836b-86afdb91a7c7.png)
mber, 0=layer order.

**Comment by shun-iwasawa** [comment]
Thank you for the information.
Hmm, I don't think it's reasonable to make some special treatment for the stroryplanner-style file name.
One option would be to enable users to specify (in preferences) any file name formats by using regular expression which will NOT be treated as sequential.. Just an idea.

**Comment by matitanimata** [comment]
![2018-06-21 2](https://user-images.githubusercontent.com/20012515/41704730-553faec6-7537-11e8-9be4-6c83dde8e417.png)

If I can give a suggestion I think that an option that enable/disable the image sequence recognition would be the simplest solution, like in the import window of adobe premiere for example.

**Comment by RodneyBaker** [comment]
@matitanimata I like your suggestion:  

> an option that enable/disable the image sequence recognition would be the simplest solution

I routinely run into problems with OpenToonz not liking the naming of files.
Of course it's not going to be ideal to turn off OT's ability to recognize sequences so a deeper dive into how files are being processed is desirable.  

Perhaps in the near future we can sit down and just run through the current process, document that thoroughly and then optimize.

**Comment by matitanimata** [comment]
for me is still a real issue because affects my pipeline. Storyplanner for Opentoonz could do the same things that toonboom storyboard actually do for Harmony. I's a pity :(

**Comment by matitanimata** [comment]
@RodneyBaker unfortunately Digital Video, for what I know now, wants to keep storyplanner as it is, so I think they won't make it open-source. In my opinion, and it is what I'm suggesting them, they should improve storyplanner and make it work in a better way with OT, I think it could be interesting for a professional use and may be we could get more studios interested in using an OT based pipeline (even if they should pay something for storyplanner, but it is quite cheap).

**Comment by matitanimata** [comment]
I would be very happy if storyplanner will become opensource, I agree with you, it needs some improvements, but if you use toonz or opentoonz it is still a valid tool for preproduction.

**Comment by RodneyBaker** [comment]
@artistteacher  Now you've got me curious if maybe that's the loading process that uses the Preference RegEx where filename formatting can be set manually.  That Regex doesn't appear to be used in the other loading/importing processes:  

Also, I hadn't caught the additional requirement for renaming the project xml file.
If I remember correctly that is a separate issue that can lead to broken projects.

**Comment by matitanimata** [comment]
finally I have found a workaround: open the tnz file with block note, find all the png mentioned and remove all the "_" from the png filename, then, in the extras folder, rename all the png files removing all the "_" 
In this way OT opens the scene without problem. It takes some time but it's worth it.

**Comment by artisteacher** [comment]
@matitanimata - when you are editing the tnz file, have you tried using find & replace? You can often replace with a delete (ie. press the delete key in the replace field) which can save a lot of time. You couldn't replace all, since underscores are used in several areas of the Toonz file. However, it might still speed things up.

**Comment by matitanimata** [comment]
@artisteacher - yes, but as you told, there are a lot of underscores in the tnz file, so the fastest way is to use FIND for "png" and then delete manually the underscores. I've asked digital video to make a little script to fix this on their side. 
I'm dreaming of an "openstoryplanner", with some adjustments and getting some features from OT like the mypaint brushes and the krita assistants would be an awesome software. I really hope that it can happen.

**Comment by RodneyBaker** [comment]
An interesting thing about Opentoonz 'Rename' command is that selecting an image in a sequence that is not recognized by Opentoonz and then using the Right CLick > Rename process tends to copy the sequence into a new correctly named sequence.  This may be well worth the time to investigate as it's a quite remedy in many cases.

To try:
- Locate a sequence of images that Opentoonz does not recognize as sequential and select the first image of the sequence.
- Right click and choose Rename
- Enter a new name (but do not add any numbering as Opentoonz will supply that)
- Apply
See if that does not make a new sequence of recognized images.

**Comment by artisteacher** [comment]
Ooh - nice! That's so much easier than manually renaming.

**Comment by manongjohn** [comment]
I remember a Discord user was having an issue loading a TNZ file created by Storyplanner.  I did look into it and determined the cause.  I had plan on fixing it for any application that creates TNZ files outside of OT. I probably got distracted by something else before I could actually work on it.

My findings:

The problem is the filenames stored in a TNZ  are being treated as if they were being imported for the first time.  That is to say that sequenced file detection is being performed on files that are in the file.  In my opinion, once a filename is stored in the scene file, sequence detection should be limited to certain conditions.

Sequenced files are normally stored as '..' or '_.' in a TNZ file, so when reading in the scene, our processes knows to expand it automatically.  Other filenames in the scene file should just be read as is, but due to our logic, stuff that may look like a file sequence is treated incorrectly as such.  So when it tries to find the file, it removes what it thinks is the seq# information and fails to find the original file.

The potential fix:

If a filename is read from a TNZ file it should be treated as is, no sequence detection should be applied to it, unless it follows the normal sequence naming convention of '..' or '_.'. 

If someone wants to tackle it before I do, let me know.

**Comment by RodneyBaker** [comment]
A likely fix for this issue has been submitted for Tahoma2D and the artifacts can be tested for it there.
REF:  https://github.com/tahoma2d/tahoma2d/pull/597/files

**Comment by RodneyBaker** [comment]
Related to #1928

**Comment by RodneyBaker** [comment]
@matitanimata shared a script that should resolve this problem with sharing files between StoryPlanner and Opentoonz.

I'm pretty sure the script can be shared but won't post publicly here as I don't have specific approval to share the files.

If you need the script however post your interest here and we'll make sure you have access.

**Comment by matitanimata** [comment]
No problem sharing the script here as well. The script is designed to rename the PNG files exported from Storyplanner so that they are readable by OpenToonz. The only thing to keep in mind is that it was devised to work with the project settings we used in our studio. In those settings, the "scenes" folder contained not only the .tnz files but also subfolders with the same names as the scenes, which held the corresponding "drawings" and "extras" folders. This setup was the easiest way to send scenes to various animators without losing any elements.

[script.zip](https://github.com/user-attachments/files/18540644/script.zip)
